src/codegen.cpp

```diff
@@ -2495,7 +2504,7 @@ static bool emit_builtin_call(jl_cgval_t *ret, jl_value_t *f, jl_value_t **args,
             if (jl_is_leaf_type(tp0)) {
                 emit_expr(args[1], ctx);
                 assert(jl_is_datatype(tp0));
-                *ret = mark_julia_type(ConstantInt::get(T_size, jl_datatype_nfields(tp0)), false, jl_long_type, ctx);
+                *ret = mark_julia_const(jl_box_long(jl_datatype_nfields(tp0)));
```

Comment:
Yeah, something like that should be fine (like the one in `ast.c`). We might also want to split the `roots` array and special case the codegen one in the serializer so that it doesn't get unnecessarily sent to the worker and cause issues like https://github.com/JuliaLang/julia/issues/15766